modules/creating-cluster-with-aws-kms-key.adoc

```diff
@@ -0,0 +1,211 @@
+// Module included in the following assemblies:
+//
+// * monitoring/enabling-monitoring-for-user-defined-projects.adoc
+
+:_content-type: PROCEDURE
+[id="creating-cluster-with-aws-kms-key"]
+= Creating a ROSA cluster using a custom AWS KMS key
+
+You can create a {product-title} (ROSA) cluster with a customer-provided KMS key that is used to encrypt both node root volumes and persistent volumes for mounted EBS claims.
+
+.Procedure
+
+. Create a custom AWS customer-managed KMS key by running the following command:
++
+[source,terminal]
+----
+$ KMS_ARN=$(aws kms create-key --region $AWS_REGION --description 'Custom ROSA Encryption Key' --tags TagKey=red-hat,TagValue=true --query KeyMetadata.Arn --output text)
+----
++
+This command saves the Amazon Resource Name (ARN) output of this custom key for further steps.
++
+[NOTE]
+====
+Customers provide the `--tags TagKey=red-hat,TagValue=true` that are required for a customer KMS key.
+====
+
+
+. Verify the KMS key has been created by running the following command:
++
+[source,terminal]
+----
+$ echo $KMS_ARN
+----
+
+. Set your AWS account ID to an environment variable.
++
+[source,terminal]
+----
+$ AWS_ACCOUNT_ID=<aws-account>
+----
+
+. Generate the necessary key policy to allow the ROSA STS roles to access the custom KMS key. Use the following command to populate a sample policy, or create your own.
+
++
+[source,terminal]
+----
+$ EBS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-openshift-cluster-csi-drivers-ebs-cloud-credential" | cut -c -64) 
+KUBE_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kube-controller-manager" | cut -c -64)
+KMS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kms-provider" | cut -c -64)
+CAPA_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-capa-controller-manager" | cut -c -64) <1>
```

Comment:
The value of operator roles arn is automatically adapted to the length of the OPERATOR_ROLES_PREFIX.It is not fixed.

I think we can only give an example in rosa-key-policy.json and reminder the operator  roles arn value is not fixed and can check it via "rosa list operator-roles --prefix ${OPERATOR_ROLES_PREFIX}"

Reply:
replaced line 31:
----
cat << EOF > rosa-key-policy.json
{
  "Version": "2012-10-17",
  "Id": "key-rosa-policy-1",
  "Statement": [
	{
  	"Sid": "Enable IAM User Permissions",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:root"
  	},
  	"Action": "kms:*",
  	"Resource": "*"
	},
	{
  	"Sid": "Allow ROSA use of the key",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": [
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Support-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Installer-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Worker-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-ControlPlane-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${EBS_ROLE}"

    	]
  	},
  	"Action": [
    	"kms:Encrypt",
    	"kms:Decrypt",
    	"kms:ReEncrypt*",
    	"kms:GenerateDataKey*",
    	"kms:DescribeKey"
  	],
  	"Resource": "*"
	},
	{
  	"Sid": "Allow attachment of persistent resources",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": [
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Support-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Installer-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Worker-Role",
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-ControlPlane-Role",AWS_ACCOUNT_ID
      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${EBS_ROLE}"
    	]
  	},
  	"Action": ["kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant"],
  	"Resource": "*",
  	"Condition": {
    	"Bool": {
      	"kms:GrantIsForAWSResource": "true"
    	}
  	}
	},
	{
  	"Sid": "ROSA Installer Permissions",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Installer-Role"
  	},
  	"Action": [
    	"kms:CreateGrant",
    	"kms:DescribeKey",
    	"kms:GenerateDataKeyWithoutPlaintext"
  	],
  	"Resource": "*"
	},
	{
  	"Sid": "ROSA KubeControllerManager Permissions",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${KUBE_ROLE}"
  	},
  	"Action": "kms:DescribeKey",
  	"Resource": "*"
	},
	{
  	"Sid": "ROSA KMS Provider Permissions",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${KMS_ROLE}"
  	},
  	"Action": ["kms:Encrypt", "kms:Decrypt", "kms:DescribeKey"],
  	"Resource": "*"
	},
	{
  	"Sid": "ROSA NodeManager Permissions",
  	"Effect": "Allow",
  	"Principal": {
    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:role/${CAPA_ROLE}"
  	},
  	"Action": [
    	"kms:DescribeKey",
    	"kms:GenerateDataKeyWithoutPlaintext",
    	"kms:CreateGrant"
  	],
  	"Resource": "*"
	}
  ]
}
EOF
----

with

Reply:
{
  "Version": "2012-10-17",
  "Id": "key-rosa-policy-1",
  "Statement": [
  {
              "Sid": "Enable IAM User Permissions",
              "Effect": "Allow",
              "Principal": {
                  "AWS": "arn:aws:iam::<aws_account_id>:root"
              },
              "Action": "kms:*",
              "Resource": "*"
          },
        {
              "Sid": "Installer Permissions",
              "Effect": "Allow",
              "Principal": {
                  "AWS": "arn:aws:iam::335462696174:role/ManagedOpenShift-HCP-ROSA-Installer-Role"
              },
              "Action": [
                  "kms:CreateGrant",
                  "kms:DescribeKey",
                  "kms:GenerateDataKeyWithoutPlaintext"
              ],
              "Resource": "*"
          },
          {
              "Sid": "ROSA KubeControllerManager Permissions",
              "Effect": "Allow",
              "Principal": {
                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-kube-controller-manager"

              },
              "Action": "kms:DescribeKey",
              "Resource": "*"
          },
          {
              "Sid": "ROSA KMS Provider Permissions",
              "Effect": "Allow",
              "Principal": {
                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-kms-provider"
              },
              "Action": [
                  "kms:Encrypt",
                  "kms:Decrypt",
                  "kms:DescribeKey"
              ],
              "Resource": "*"
          },
          {
              "Sid": "ROSA NodeManager Permissions",
              "Effect": "Allow",
              "Principal": {
                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-capa-controller-manager"
              },
              "Action": [
                  "kms:DescribeKey",
                  "kms:GenerateDataKeyWithoutPlaintext",
                  "kms:CreateGrant"
              ],
              "Resource": "*"
          }
      ]
  }